### PR TITLE
Generate the TOC during the disassembly step

### PR DESCRIPTION
**Pull request by m1yag1**
Generate the TOC during the disassembly step

### CODE REVIEW

File: script/disassemble.py [hunk 1]
```diff
@@ -1,18 +1,32 @@
...
 with open(baked_file, "rb") as file:
+    html_root = etree.parse(file)
     binder = reconstitute(file)
 
+nav = html_root.xpath("//xhtml:nav", namespaces=HTML_DOCUMENT_NAMESPACES)
+
+toc_maker = ElementMaker(namespace=None,
+                         nspam={None: "http://www.w3.org/1999/xhtml"})
+toc = toc_maker.html(E.head(E.title("Table of Contents")),
+                     E.body(nav))
```
Comment (rnathuji) [inline script/disassemble.py]: @m1yag1 - Passing the nav data this way seems to be problematic (?) when I try to run as I get a `TypeError: bad argument type: list([<Element {http://www.w3.org/1999/xhtml}nav at 0x7f5b525c4508>])` .

**Comment by rnathuji** [comment]
@m1yag1 - Just FYI, as part of #850 we'll be adding some basic testing for these various steps / scripts. You'll be merging this to master first anyways, so if it makes sense to you we can incorporate a basic test case for this as part of #850 since we'll have to merge with your changes anyways (e.g. to be sure we don't break you =) ).

File: script/disassemble.py [hunk 2]
```diff
@@ -1,18 +1,32 @@
...
 baked_file = (in_dir / "collection.baked.xhtml").resolve(strict=True)
 
 with open(baked_file, "rb") as file:
+    html_root = etree.parse(file)
     binder = reconstitute(file)
 
+nav = html_root.xpath("//xhtml:nav", namespaces=HTML_DOCUMENT_NAMESPACES)
+
+toc_maker = ElementMaker(namespace=None,
+                         nspam={None: "http://www.w3.org/1999/xhtml"})
```
[the remark this answers is not in the document]
Comment (m1yag1) [reply script/disassemble.py]: yep it should be. lol

File: script/disassemble.py (hunk 1 above)
Comment (m1yag1) [reply script/disassemble.py]: interesting... it ran fine for me.

Comment (m1yag1) [reply script/disassemble.py]: now that i can execute tasks in concourse based on the code in my local machine i'll try it out and see.

File: script/disassemble.py [hunk 3]
```diff
@@ -1,18 +1,32 @@
...
 
+nav = html_root.xpath("//xhtml:nav", namespaces=HTML_DOCUMENT_NAMESPACES)
+
+toc_maker = ElementMaker(namespace=None,
+                         nsmap={None: "http://www.w3.org/1999/xhtml"})
+toc = toc_maker.html(E.head(E.title("Table of Contents")),
+                     E.body(nav))
+
+with open(f"{out_dir}/collection.toc.xhtml", "wb") as out:
+    out.write(etree.tostring(toc, encoding="utf8", pretty_print=True))
```
Comment (tomjw64) [inline script/disassemble.py]: pretty_printing typically comes with size/bandwidth drawbacks. What is the benefit of pretty_printing in this case?

Comment (m1yag1) [reply script/disassemble.py]: I saw it used within the code in many places. That's the only reason.

Comment (m1yag1) [reply script/disassemble.py]: @tomjw64 For example, in the same function you used to output the disassembled xhtml file it uses `pretty_print`. https://github.com/openstax/cnx-epub/blob/master/cnxepub/formatters.py#L64

Comment (m1yag1) [reply script/disassemble.py]: and here: https://github.com/openstax/cnx-epub/blob/master/cnxepub/formatters.py#L307-L310

Comment (m1yag1) [reply script/disassemble.py]: sooo.. is that a good reason? "That's the way it's always been done". It's kinda precedent at this point so I stuck with it. I can remove it if you think that would be best. I don't have a strong preference either way.

Comment (tomjw64) [reply script/disassemble.py]: Good enough for me!

**Pull request merged**